eth/interop/interop.go

```diff
@@ -0,0 +1,57 @@
+package interop
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/core/types/interoptypes"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type InteropClient struct {
+	mu       sync.Mutex
+	client   *rpc.Client
+	endpoint string
+	closed   bool // don't allow lazy-dials after Close
+}
+
+// maybeDial dials the endpoint if it was not already.
+func (cl *InteropClient) maybeDial(ctx context.Context) error {
+	cl.mu.Lock()
+	defer cl.mu.Unlock()
+	if cl.closed {
+		return errors.New("client is closed")
+	}
+	if cl.client != nil {
+		return nil
+	}
+	rpcClient, err := rpc.DialContext(ctx, cl.endpoint)
+	if err != nil {
+		return err
+	}
+	cl.client = rpcClient
+	return nil
+}
+
+func (cl *InteropClient) Close() {
+	cl.mu.Lock()
+	defer cl.mu.Unlock()
+	if cl.client != nil {
+		cl.Close()
```

Comment:
This needs to call `cl.client.Close()` instead of `cl.Close()`. I think a rebase should fix this.

Reply:
rebased :+1: